### ISSUE TITLE
"Use discrete graphics" option in settings works backwards on desktops

### ISSUE THREAD

**Issue by brndd** [lutris/lutris#4416]
On a typical desktop system that has a discrete and an integrated GPU with displays plugged into the discrete GPU, the "Use discrete graphics" option under "System options" works backwards; toggling it on will cause games to use the integrated graphics.

This is because that option, when enabled, sets `DRI_PRIME=1`, which [according to the Mesa documentation](https://gitlab.freedesktop.org/mesa/mesa/-/blob/main/docs/envvars.rst#id79) "selects the first non-default GPU". On a system where the display is connected to the discrete GPU, the first non-default GPU is the iGPU. Since this setting is enabled by default, users will get very poor performance unless they realize to disable it as the game runs on their iGPU instead of their discrete graphics.

**Comment by danieljohnson2** [comment]
I've experimented with this, trying to write a check to see if the default GPU is NVIDIA or AMD via lspci, but I see no obvious way to know which graphics adapter is the default one. Is there some way to tell this?

If I could tell, I could detect if it were NVIDIA or AMD and suppress DRI_PRIME, but without that it'd be unreliable.

**Comment by brndd** [comment]
> If I could tell, I could detect if it were NVIDIA or AMD and suppress DRI_PRIME, but without that it'd be unreliable.

What about `glxinfo | grep "OpenGL renderer"`? 

```
$ DRI_PRIME=0 glxinfo | grep "OpenGL renderer"
OpenGL renderer string: AMD Radeon RX 6700 XT (navy_flounder, LLVM 14.0.0, DRM 3.46, 5.18.13-200.fc36.x86_64)
$ DRI_PRIME=1 glxinfo | grep "OpenGL renderer"
OpenGL renderer string: Mesa Intel(R) Graphics (ADL-S GT1)
```

That being said, Intel is now making dedicated GPUs too (and AMD makes integrated ones) so it may not be as simple as detecting the manufacturer.

**Comment by danieljohnson2** [comment]
That's interesting. Maybe we could replace lspci with glxinfo, but that seems a big change for this issue.

The "renderer string" does *not* match the PCI device description that lspci gives, so I can't cross reference simply. I'd hope to get this information out of lspci.

**Comment by strycore** [comment]
Please test with 0.5.13, I believe this is fixed.

**Comment by brndd** [comment]
Does not appear to be fixed on lutris-0.5.13-2.fc38.src.rpm from the Fedora repos.

The settings below, which I believe are the default, make the game (Vintage Story in this case) run on the iGPU with abysmal performance. Turning off "Use discrete graphics" solves the issue.

![2023-06-21T01:24:49](https://github.com/lutris/lutris/assets/5822375/821ceb40-a8cd-4e68-b650-729cedc30f77)

**Comment by strycore** [comment]
I think this happens because the current code thinks "discrete GPU = Nvidia"

**Comment by brndd** [comment]
Well, I have a discrete AMD GPU and some people even have discrete Intel GPUs, so I think this should be reopened.

**Comment by hatani** [comment]
Same issue confirmed on my side, on Fedore + Gnome you can rightclick pinned Lutris and 'run using discreet graphics' which seems to work luckily.

**Comment by al12gamer** [comment]
Another easy way to fix this would to not use a CPU with integrated graphics.

**Comment by brndd** [comment]
There is a program called `switcherooctl` that can show what the user's default GPU is:

```
$ switcherooctl list
Device: 0
  Name:        Advanced Micro Devices, Inc. [AMD®/ATI] Navi 22 [Radeon RX 6700/6700 XT/6750 XT / 6800M/6850M XT] (Radeon RX 6700 XT Mech 2X 12G [MSI])
  Default:     yes
  Environment: DRI_PRIME=pci-0000_03_00_0

Device: 1
  Name:        Intel Corporation AlderLake-S GT1
  Default:     no
  Environment: DRI_PRIME=pci-0000_00_02_0
```

There is an open MR (that may never get merged) into switcherooctl to add discrete GPU detection: https://gitlab.freedesktop.org/hadess/switcheroo-control/-/merge_requests/69

However, switcherooctl does not look to be actively developed anymore so this MR may go unmerged. The logic used in it could possibly be used in Lutris; there are also MRs open for KDE and Gnome that seek to do similar things: https://invent.kde.org/frameworks/kio/-/merge_requests/1556

**Comment by strycore** [comment]
"Use discrete graphics" has been removed from Lutris. This was half implemented and didn't work properly. The next Lutris release will simply have a dropdown with a GPU selection. There will be no concept of primary GPU. I improved multi-GPU support because I coded this on a desktop computer using a Radeon 7800XT and a RTX 3060. Under no circumstance would I want Lutris to default to the Nvidia GPU as the "main" GPU. 
I still don't have a laptop with dual GPUs to test things but at this point, I no longer believe it's necessary.
